spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/pubsub/PubSubHealthIndicator.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2017-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.autoconfigure.pubsub;
+
+import org.springframework.boot.actuate.health.AbstractHealthIndicator;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.cloud.gcp.pubsub.PubSubAdmin;
+
+/**
+ * Default implemenation of
+ * {@link org.springframework.boot.actuate.health.HealthIndicator} for Pubsub. Checks if
+ * the credentials being used for the project can invoke PubSub APIs by calling
+ * {@link PubSubAdmin} listTopics() method.
+ *
+ * @author Vinicius Carvalho
+ */
+public class PubSubHealthIndicator extends AbstractHealthIndicator {
+
+	private PubSubAdmin pubSubAdmin;
+
+	public PubSubHealthIndicator(PubSubAdmin pubSubAdmin) {
+		super("Failed to connect to PubSub APIs. Check your credentials and verify you have proper access to the service.");
+		this.pubSubAdmin = pubSubAdmin;
+	}
+
+	@Override
+	protected void doHealthCheck(Health.Builder builder) throws Exception {
+		try {
+			pubSubAdmin.listTopics();
```

Comment:
I think it might be possible for an account to have access to publish to topics but not to list them.
Also, this might be an expensive operation if there are many topics.
A possible idea that we applied in other places is to do a simple operation like trying to read from a non-existing subscription, and to declare HEALTH_UP when you get an error message that says something like subscription doesn't exist. WDYT?

Reply:
Thank you @meltsufin, agreed on the suggestion. Will modify this and submit the changes